server/sonar-server/src/main/java/org/sonar/server/organization/OrganizationCreation.java

```diff
@@ -61,8 +63,28 @@
    */
   OrganizationDto create(DbSession dbSession, long createUserId, NewOrganization newOrganization) throws KeyConflictException;
 
+  /**
+   * Create a new guarded organization which details are based on the login of the specified User.
+   * <p>
+   * This method create the organization and its associated elements in exactly the same was as
+   * {@link #create(DbSession, long, NewOrganization)} with the organization's details computed from the
+   * user's login:
+   * <ul>
+   *   <li>key: generated from the user's login</li>
+   *   <li>name: the user's login</li>
+   *   <li>description, url and avatar: null</li>
+   * </ul>
+   * </p>
+   *
+   * @return the created organization or empty if feature is disabled
```

Comment:
the return type if an Optional just to indicate that the method does not always create the organization